server/parameters/CoreSubcommandParser.java

```diff
@@ -71,24 +72,24 @@ public Server(CoreConfigParser configParser) {
         }
 
         @Override
-        protected ServerSubcommand.Server parse(Set<Option> options) {
-            Set<Option> auxOptions = findAuxiliaryOptions(options);
+        protected CoreSubcommand.Server parse(Set<Option> options) {
+            Set<Option> auxOptions = findAuxiliaryOptions(auxiliaryParsers, options);
             Set<Option> configOptions = excludeOptions(options, auxOptions);
             Optional<Path> configPath = Server.configPath.parse(auxOptions);
             CoreConfig config = configPath
                     .map(path -> CoreConfigFactory.config(path, configOptions, configParser))
                     .orElseGet(() -> CoreConfigFactory.config(configOptions, configParser));
-            return new ServerSubcommand.Server(debug.parse(auxOptions), help.parse(auxOptions),
+            return new CoreSubcommand.Server(debug.parse(auxOptions), help.parse(auxOptions),
                     version.parse(auxOptions), config);
         }
 
-        private Set<Option> findAuxiliaryOptions(Set<Option> options) {
+        public static Set<Option> findAuxiliaryOptions(Set<OptionParser> auxiliaryParsers, Set<Option> options) {
```

Comment:
make `static` so that the Cluster implementation can reference without re-impelementing